packages/edit-site/src/components/create-pattern-modal/index.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	TextControl,
+	Button,
+	Modal,
+	ToggleControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { store as noticesStore } from '@wordpress/notices';
+import { useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { SYNC_TYPES, USER_PATTERN_CATEGORY } from '../page-library/constants';
+
+export default function CreatePatternModal( {
+	closeModal,
+	onCreate,
+	onError,
+} ) {
+	const [ name, setName ] = useState( '' );
+	const [ syncType, setSyncType ] = useState( SYNC_TYPES.full );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+
+	const onSyncChange = () => {
+		setSyncType(
+			syncType === SYNC_TYPES.full ? SYNC_TYPES.unsynced : SYNC_TYPES.full
+		);
+	};
+
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	async function createPattern() {
+		if ( ! name ) {
+			createErrorNotice( __( 'Name is not defined.' ), {
```

Comment:
It's a little bit brief, maybe something like:
```suggestion
			createErrorNotice( __( 'Please enter a pattern name.' ), {
```

Reply:
Let's follow up with some copy review/updates.